src/autoscaler/metricscollector/cmd/metricscollector/metricscollector_test.go

```diff
@@ -21,22 +32,74 @@ var _ = Describe("MetricsCollector", func() {
 		runner = NewMetricsCollectorRunner()
 	})
 
-	JustBeforeEach(func() {
-		runner.Start()
-	})
-
 	AfterEach(func() {
 		runner.KillWithFire()
 	})
 
-	It("should start", func() {
-		Consistently(runner.Session).ShouldNot(Exit())
+	Context("when the metricscollector acquires the lock", func() {
+		BeforeEach(func() {
+			runner.StartWithoutCheck()
+			Eventually(runner.Session.Buffer, 2*time.Second).Should(gbytes.Say(runner.acquiredLockCheck))
+		})
+
+		It("should start", func() {
+			Eventually(runner.Session.Buffer(), 2*time.Second).Should(gbytes.Say(runner.startCheck))
+			Consistently(runner.Session).ShouldNot(Exit())
+		})
+	})
+
+	Context("when the metricscollector loses the lock", func() {
+		BeforeEach(func() {
+			runner.StartWithoutCheck()
+			Eventually(runner.Session.Buffer, 2*time.Second).Should(gbytes.Say(runner.acquiredLockCheck))
+			Eventually(runner.Session.Buffer(), 2*time.Second).Should(gbytes.Say(runner.startCheck))
+
+			consulRunner.Reset()
+		})
+
+		It("exits with failure", func() {
+			Eventually(runner.Session.Buffer(), 4*time.Second).Should(gbytes.Say("exited-with-failure"))
+			Eventually(runner.Session).Should(Exit(1))
+		})
+	})
+
+	Context("when the metricscollector initially does not have the lock", func() {
+		var competetingMetricsCollectorProcess ifrit.Process
+
+		BeforeEach(func() {
+			consulClient := consulRunner.NewClient()
+			logger := lager.NewLogger("metricscollector")
+			competingMetricsCollectorLock := locket.NewLock(logger, consulClient, metricscollector.MetricsCollectorLockSchemaPath(), []byte{}, clock.NewClock(), cfg.Lock.LockRetryInterval, cfg.Lock.LockTTL)
+			competetingMetricsCollectorProcess = ifrit.Invoke(competingMetricsCollectorLock)
+
+			runner.StartWithoutCheck()
+		})
+
+		It("should not start", func() {
+			Eventually(runner.Session.Buffer, 2*time.Second).Should(gbytes.Say("metricscollector.lock.acquiring-lock"))
+			Consistently(runner.Session.Buffer, 2*time.Second).ShouldNot(gbytes.Say(runner.startCheck))
+		})
+
+		Describe("when the lock becomes available", func() {
+			BeforeEach(func() {
+				ginkgomon.Kill(competetingMetricsCollectorProcess)
+				time.Sleep(10 * time.Millisecond)
+			})
+
+			It("acquires the lock and starts", func() {
+				Eventually(runner.Session.Buffer, 2*time.Second).Should(gbytes.Say(runner.acquiredLockCheck))
+				Eventually(runner.Session.Buffer(), 2*time.Second).Should(gbytes.Say(runner.startCheck))
+				Consistently(runner.Session).ShouldNot(Exit())
+			})
+
+		})
 	})
 
 	Context("with a missing config file", func() {
 		BeforeEach(func() {
 			runner.startCheck = ""
 			runner.configPath = "bogus"
+			runner.Start()
```

Comment:
runner.Start(), here is doing what StartWithoutCheck is doing as startCheck = ""